platform_channels/lib/src/platform_image_demo.dart

```diff
@@ -0,0 +1,68 @@
+import 'dart:typed_data';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:platform_channels/src/image_basic_message_channel.dart';
+
+/// Demonstrates how to use [BasicMessageChannel] to get an image from platform.
+///
+/// The widget uses [Image.memory] to display the image obtained from the
+/// platform.
+class PlatformImageDemo extends StatefulWidget {
+  @override
+  _PlatformImageDemoState createState() => _PlatformImageDemoState();
+}
+
+class _PlatformImageDemoState extends State<PlatformImageDemo> {
+  Uint8List imageData;
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Platform Image Demo'),
+      ),
+      body: Builder(
+        builder: (context) {
+          return Center(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                Container(
```

Comment:
Have you seen FractionallySizedBox? I believe you can use it to get the look you want without the need for a MediaQuery.

https://github.com/flutter/flutter/blob/f7f019aa9410f484055aee7207cc51a61ad95ba7/packages/flutter/lib/src/widgets/basic.dart#L2300